### PR TITLE
Switch to use metadata API instead of SQLAlchemy

### PR DESCRIPTION
**Pull request by bilalebi**
> ~NB1: /!\ This PR can be merged only after https://github.com/Ensembl/ensembl-metadata-api/pull/31 is~ DONE.

> ~NB2: Before merging, don't forget to update `requirements.txt` to point to the latest `ensembl-metadata-api`~ DONE.

### JIRA Ticket
https://www.ebi.ac.uk/panda/jira/browse/EA-1087

### Changes
* Switch to use metadata API instead of SQLAlchemy.
* Did some refactoring (of datasets-related code, statistics, assembly info and species).
* Split `service.py` file into 3 files and move them to `/grpc/` directory/module.
* Split test file.
* Fixed broken tests.

# Related PR
https://github.com/Ensembl/ensembl-metadata-api/pull/31

### CODE REVIEW

File: src/ensembl/__init__.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#  See the NOTICE file distributed with this work for additional information
```
Comment (marcoooo) [inline src/ensembl/__init__.py:1]: This file shouldn't exist to allow namespace packaging

File: src/ensembl/production/__init__.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#  See the NOTICE file distributed with this work for additional information
```
Comment (marcoooo) [inline src/ensembl/production/__init__.py:1]: This file shouldn't exist to allow namespace packaging

File: src/ensembl/production/metadata/__init__.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#  See the NOTICE file distributed with this work for additional information
```
Comment (marcoooo) [inline src/ensembl/production/metadata/__init__.py:1]: This file shouldn't exist to allow namespace packaging

**Review by marcoooo — CHANGES_REQUESTED** [review]

File: src/ensembl/production/metadata/__init__.py
Comment (marcoooo) [inline src/ensembl/production/metadata/__init__.py:1]: Just to make sure, was this file retired? For compatibility with other namespaced packages.

Comment (bilalebi) [reply src/ensembl/production/metadata/__init__.py:1]: > Just to make sure, was this file retired? For compatibility with other namespaced packages.

Yes, as you suggested [in this comment](https://github.com/Ensembl/ensembl-metadata-service/pull/25#discussion_r1318653936).
I don't mind putting it back as I don't see much difference (Am I missing something?)

Comment (bilalebi) [reply src/ensembl/production/metadata/__init__.py:1]: @marcoooo, it's not there, as you can see here: https://github.com/Ensembl/ensembl-metadata-service/tree/chore/switch_to_md_api/src/ensembl/production/metadata

The `__init__.py` doesn't exist, even though it says _empty file_ as you described 🤷‍♂️

**Review by marcoooo — APPROVED** [review]

**Pull request merged**